rust/tools/authorization-logic/src/souffle/lowering_ast_datalog.rs

```diff
@@ -349,7 +492,15 @@ impl LoweringToDatalogPass {
         };
         dlir_assertions.append(&mut dlir_queries);
         dlir_assertions.push(dummy_assertion);
+
+        // type declarations
```

Comment:
Replicate the comment about how this should be called after body is emitted?

Reply:
ok